dec_term: must use unsigned for BINARY_EXT size

mikpe opened the pull request:
Fixes #6393

The fix needs to be backported to OTP-25 and OTP-24 (OTP-23 doesn't crash but throws an IMO bogus error instead).

There may be similar bugs elsewhere in this code, but I haven't investigated that in detail (yet).

sverker commented:
@mikpe I rebased your commit on OTP-23.3.4, added some more fixes and tests, and then merged it forward to 24.3.4 to solve some conflicts. And then force push to your repo to keep this PR going. Hope that was ok.

mikpe commented:
> @mikpe I rebased your commit on OTP-23.3.4, added some more fixes and tests, and then merged it forward to 24.3.4 to solve some conflicts. And then force push to your repo to keep this PR going. Hope that was ok.

@sverker no problem
